misc/deprecated/domain/builders/atrium-domain-builders-common/src/main/kotlin/ch/tutteli/atrium/domain/builders/utils/VarArgHelper.kt

```diff
@@ -29,12 +30,51 @@ interface VarArgHelper<out T> {
 }
 
 /**
- * Transforms the given [iterable] to `Pair<T, Array<out T>>` with the intend that it can be easily used for a function
+ * Transforms the given [IterableLike] to `Pair<T, Array<out T>>` with the intend that it can be easily used for a function
  * requiring `T, vararg T`
  *
  * @throws IllegalArgumentException in case the iterable is empty.
  */
-inline fun <reified T> toVarArg(iterable: Iterable<T>): Pair<T, Array<out T>> {
+inline fun <reified T> toVarArg(iterableLike: IterableLike): Pair<T, Array<out T>> {
+    when (iterableLike) {
+        is Iterable<*> -> {
+            return iterableToPair(iterableLike.map { it as T })
+        }
+        is Sequence<*> -> {
+            return iterableToPair(iterableLike.map { it as T }.asIterable())
+        }
+        is Array<*> -> {
+            return iterableToPair(iterableLike.map { it as T }.asIterable())
```

Comment:
No need for `asIterable` (also below) as map already transforms the Array into a List which is a subtype of Iterable
```suggestion
            return iterableToPair(iterableLike.map { it as T })
```